javaparser-testing/src/test/java/com/github/javaparser/utils/SourceZipTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2007-2010 Júlio Vilmar Gesser.
+ * Copyright (C) 2011, 2013-2016 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.utils;
+
+import static com.github.javaparser.utils.CodeGenerationUtils.f;
+import static com.github.javaparser.utils.TestUtils.download;
+import static com.github.javaparser.utils.TestUtils.temporaryDirectory;
+import static java.util.Comparator.comparing;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.junit.Test;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.Problem;
+import com.github.javaparser.ast.validator.Java9Validator;
+
+public class SourceZipTest {
+
+    private final static String PROJECT_NAME = "JavaParser";
+    private final static String PROJECT_TEST_RESULT_FILE = PROJECT_NAME + "_Test_Results.txt";
+    private final static String ZIP_URL = "https://github.com/javaparser/javaparser/archive/master.zip";
+    private final static String ZIP_DOWNLOAD_DIR = PROJECT_NAME + "_Download";
+    private final static String ZIP_NAME = "master.zip";
+
+    @Test
+    public void parseProjectZip() throws IOException {
+        Path workdir = CodeGenerationUtils.mavenModuleRoot(SourceZipTest.class)
+                .resolve(Paths.get(temporaryDirectory(), ZIP_DOWNLOAD_DIR));
+        workdir.toFile().mkdirs();
+        Path zipPath = workdir.resolve(ZIP_NAME);
+        if (Files.notExists(zipPath)) {
+            Log.info("Downloading " + PROJECT_NAME);
+            download(new URL(ZIP_URL), zipPath);
+        }
+        bulkTest(new SourceZip(zipPath), PROJECT_TEST_RESULT_FILE);
+    }
+
+    private void bulkTest(SourceZip sourceZip, String testResultsFileName) throws IOException {
+        sourceZip.setJavaParser(new JavaParser(new ParserConfiguration().setValidator(new Java9Validator())));

```

Comment:
What you could do instead is change the existing BulkParseTest like this and call that the test. You will have to figure out #941 then though :-(

Reply:
I mean - this class is way too huge to be called a unit test, it writes a file with parse problems so that the parser can be verified.